cs-utilities/src/test/java/io/cloudslang/content/utilities/actions/DefaultIfEmptyTest.java

```diff
@@ -0,0 +1,103 @@
+ */
+public class DefaultIfEmptyTest {
+
+    private final DefaultIfEmptyAction d = new DefaultIfEmptyAction();
+
+    @Test
+    public void testWithBlankString() {
+        final Map<String, String> result = d.execute("   ", "string", "  ");
+        assertEquals("0", result.get(RETURN_CODE));
+        assertEquals("string", result.get(RETURN_RESULT));
+    }
+
+    @Test
+    public void testWithEmptyString() {
+        final Map<String, String> result = d.execute("", "string", "");
+        assertEquals("0", result.get(RETURN_CODE));
+        assertEquals("string", result.get(RETURN_RESULT));
+    }
+
+    @Test
+    public void testWithTrimString() {
+        final Map<String, String> result = d.execute("", "string", "astring");
+        assertEquals("-1", result.get(RETURN_CODE));
+        assertEquals("Failure", result.get(RETURN_RESULT));
+    }
+
+    @Test
+    public void testWithTrimInitialString() {
+        final Map<String, String> result = d.execute("initial", "string", "astring");
+        assertEquals("-1", result.get(RETURN_CODE));
+        assertEquals("Failure", result.get(RETURN_RESULT));
+    }
+
+    @Test
+    public void testHavingInitialString() {
+        final Map<String, String> result = d.execute("initialstr", "string", "");
+        assertEquals("0", result.get(RETURN_CODE));
+        assertEquals("initialstr", result.get(RETURN_RESULT));
+    }
+
+    @Test
+    public void testWithTrimTrue() {
+        final Map<String, String> result = d.execute("   ", "string", "true");
+        assertEquals("0", result.get(RETURN_CODE));
+        assertEquals("string", result.get(RETURN_RESULT));
+    }
+
+    @Test
+    public void testWithTrimFalse() {
+        final Map<String, String> result = d.execute("", "string", "false");
+        assertEquals("0", result.get(RETURN_CODE));
+        assertEquals("string", result.get(RETURN_RESULT));
+    }
+
+    @Test
+    public void testHavingInitialStringAndTrimTrue() {
+        final Map<String, String> result = d.execute("initialstr", "string", "true");
+        assertEquals("0", result.get(RETURN_CODE));
+        assertEquals("initialstr", result.get(RETURN_RESULT));
+    }
+
+    @Test
+    public void testHavingInitialStringAndTrimFalse() {
+        final Map<String, String> result = d.execute("initialstr", "string", "false");
+        assertEquals("0", result.get(RETURN_CODE));
+        assertEquals("initialstr", result.get(RETURN_RESULT));
+    }
+
+    @Test
+    public void testHavingTrimNull() {
+        final Map<String, String> result = d.execute("", "string", null);
+        assertEquals("0", result.get(RETURN_CODE));
+        assertEquals("string", result.get(RETURN_RESULT));
+    }
+
+    @Test
+    public void testHavingInitialStringNull() {
+        final Map<String, String> result = d.execute(null, "string", "false");
+        assertEquals("0", result.get(RETURN_CODE));
+        assertEquals("string", result.get(RETURN_RESULT));
+    }
```

Comment:
create a test where "false" is: 
- typed with spaces at the end/ beginning
- or some like this: "tRUe" ... be creative